src/app/app.component.ts

```diff
@@ -0,0 +1,111 @@
+import { HttpClient } from '@angular/common/http';
+import { AfterViewInit, Component, ViewChild } from '@angular/core';
+import { MatSort } from '@angular/material/sort';
+import { MatTable, MatTableDataSource } from '@angular/material/table';
+import { tap } from 'rxjs';
+import { FilterValue } from './filter/filter.component';
+import { typeColors } from './legend/legend.component';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.scss']
+})
+export class AppComponent implements AfterViewInit {
+  private currentData: Row[] = [];
+  private filterValue: FilterValue = {
+    currentPage: 0,
+    entriesPerPage: 20
+  };
+
+  readonly title = 'BG Queue Abuser Viewer';
+  readonly displayedColumns: readonly (keyof Row)[] = Object.freeze([
+    'position',
+    'guid',
+    'account',
+    'name',
+    'level',
+    'type',
+    'datetime'
+  ]);
+  readonly typeColors = typeColors;
+
+  loading = true;
+  dataSource: MatTableDataSource<Row> = new MatTableDataSource<Row>(this.currentData);
+
+  @ViewChild(MatTable) table!: MatTable<Row>;
+  @ViewChild(MatSort) sort!: MatSort;
+
+  constructor(private http: HttpClient) {}
+
+  ngAfterViewInit() {
+    this.dataSource.sort = this.sort;
+
+    this.getDataAndRefresh();
+  }
+
+  private getData() {
+    return this.http
+      .get(
+        `api/characters/battleground_deserters/${this.filterValue.entriesPerPage}?from=${
+          this.filterValue.currentPage * this.filterValue.entriesPerPage
+        }${this.filterValue.nameFilter ? '&name=' + this.filterValue.nameFilter : ''}`
+      )
+      .pipe(
+        tap(obj => {
+          this.currentData = (obj as APIResults[]).map((res, index) => {
+            const newRow = {
+              ...res,
+              position: index + 1 + this.filterValue.currentPage * this.filterValue.entriesPerPage
+            };
+
+            const datetime = new Date(newRow.datetime);
+            const formatter = Intl.DateTimeFormat(undefined, {
+              dateStyle: 'short',
+              timeStyle: 'long'
+            });
```

Comment:
I don't know Intl.DateTimeFormat but that `undefined` sounds weird.

Reply:
moreover, are you sure that Angular does not have already some pipes to handle this?

Reply:
> I don't know Intl.DateTimeFormat but that `undefined` sounds weird.

As this [MDN page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#locale_identification_and_negotiation) says, passing undefined means "The implementation's default locale will be used".

Reply:
> moreover, are you sure that Angular does not have already some pipes to handle this?

There is, but I didn't know it.
Still I think this is the best way of converting the date in this specific context.
In fact applying the pipe would make the template less generic.

Reply:
ok, I know about the undefined thing, but may there is an elegant way to do it

but, that's fine also in this way